### PR TITLE
VW version requirement and documentation on config_constraints vs metric_constraints

### PR DESCRIPTION
**Pull request by qingyun-wu**
## Why are these changes needed?
1. Adding VW version requirements to ensure AutoVW's compatibility with VW.
2. Add documentation on config_constraints and metric_constraints. 

## Checks

- [x] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR, or I've made sure [lint with flake8](https://github.com/microsoft/FLAML/blob/816a82a1155b4de4705b21a615ccdff67c6da379/.github/workflows/python-package.yml#L54-L59) output is two 0s.
- [x] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

File: website/docs/Use-Cases/Tune-User-Defined-Function.md [hunk 1]
```diff
@@ -265,24 +265,27 @@ A user can specify constraints on the configurations to be satisfied via the arg
...
 ```python
-flaml.tune.run(evaluation_function=evaluate_config, mode="min",
-               config=config_search_space,
-               metric_constraints=[("score", "<=", 0.4)],...)
+analysis = tune.run(...,
+    metric_constraints = [("training_cost", "<=", 1)]),
 ```
 
+#### **`config_constraints` vs `metric_constraints`:**
+The key difference between these two types of constraints is that the calculation of constraints in `config_constraints` **does not rely on the evaluation procedure in the metric function**. For example, when a constraint only depends on the config itself, as shown in the code example. Due to this independency, constraints in `config_constraints` will be checked before evaluation. So configurations that do not satisfy `config_constraints` will not be evaluated.
```
Comment (sonichi) [inline website/docs/Use-Cases/Tune-User-Defined-Function.md:286]: What's "metric function"? Could you rephrase it to avoid confusion?

**Review by sonichi — COMMENTED** [review]
Please make the auto checks pass.

**Review by olgavrou — APPROVED** [review]

**Review by sonichi — APPROVED** [review]

File: website/docs/Use-Cases/Tune-User-Defined-Function.md [hunk 2]
```diff
@@ -265,24 +265,27 @@ A user can specify constraints on the configurations to be satisfied via the arg
...
 ```python
-flaml.tune.run(evaluation_function=evaluate_config, mode="min",
-               config=config_search_space,
-               metric_constraints=[("score", "<=", 0.4)],...)
+analysis = tune.run(...,
+    metric_constraints = [("training_cost", "<=", 1)]),
 ```
 
+#### **`config_constraints` vs `metric_constraints`:**
+The key difference between these two types of constraints is that the calculation of constraints in `config_constraints` does not rely on the computation procedure in the evaluation function, i.e., in `evaluation_function`. For example, when a constraint only depends on the config itself, as shown in the code example. Due to this independency, constraints in `config_constraints` will be checked before evaluation. So configurations that do not satisfy `config_constraints` will not be evaluated.
```
Comment (YIWEI-CHEN) [inline website/docs/Use-Cases/Tune-User-Defined-Function.md:286]: My suggestion:
`metric_constraints` need model's output or statistic of evaluation function (e.g., training cost) and are computed in the `evaluation_function`. `config_constraints` only depend on the config itself, as shown in the model size example. `config_constraints` does not rely on the computation procedure in `evaluation_function`. Due to this independency, `config_constraints` will be checked before evaluation. The configurations that do not satisfy `config_constraints` will not be used in `evaluation_function`.

**Review by YIWEI-CHEN — COMMENTED** [review]
My suggestion for `config_constraints` vs `metric_constraints`:
`metric_constraints` need model's output (e.g., Precision) or statistic of evaluation function (e.g., training cost). They are computed in the `evaluation_function`. `config_constraints` only depend on the config itself, as shown in the model size example. `config_constraints` does not rely on the computation procedure in evaluation_function, so config_constraints will be checked before evaluation. The configurations that do not satisfy config_constraints are not used in `evaluation_function`.
